### PR TITLE
Sprout guide to locking unlocking rooms

### PR DESCRIPTION
**Pull request by zspencer**
See: https://github.com/zinc-collective/convene/issues/40

- We can now use `.md` and `.markdown` files as erb partials.
- There's now a "breadcrumb" component for allowing folks to go back.
- The Feature model is starting to hold more responsibility.
Screen Shot 2021-01-15 at 4 33 52 PM
Screen Shot 2021-01-15 at 4 32 24 PM

### CODE REVIEW

File: convene-web/app/views/features/_locked_rooms.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+People who entered the Locked Room before may enter automatically, so long as
+the passcode has not changed since they last entered the room.
+
+## Some Caveats
+
+1. Locking a Room does not remove any Occupants already in the Room, but it will
+   require them to enter the passcode when they attempt to return to the Room.
+2. Unlocking a Room also requires the passcode. If you forget the password,
+   reach out to our support team and we'll unlock it for you.
```
Comment (zspencer) [inline convene-web/app/views/features/_locked_rooms.md:23]: @colombene - How does this read from a technical-writing "how to use a feature" perspective?

File: convene-web/app/views/features/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<nav class="breadcrumbs">
+  <span class="crumb">
+    <%= t('features.title') %>
+  </span>
+</nav>
```
Comment (zspencer) [inline convene-web/app/views/features/index.html.erb:5]: @user512 / @anaulin I _think_ a client of mine uses [this gem](https://github.com/fnando/breadcrumbs) for their breadcrumb system; which seems to work pretty well. Thoughts?

File: convene-web/app/javascript/src/custom_components.scss [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
+@import "./custom_components/form";
+@import "./custom_components/header";
 @import "./custom_components/icons";
+
+@import "./custom_components/meeting_room";
+
+@import "./custom_components/room_card";
 @import "./custom_components/room_edit";
+
 @import "./custom_components/workspaces";
```
Comment (zspencer) [inline convene-web/app/javascript/src/custom_components.scss:12]: Apologies for the noise here, I ordered them alphabetically.

File: convene-web/Gemfile [hunk 4]
```diff
@@ -11,36 +11,61 @@ gem 'dotenv-rails', groups: [:development, :test]
...
 
+# Date/Time and Internationalization
+#
+# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
+gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+
+# Performance Optimization
+#
+# Reduces boot times through caching; required in config/boot.rb
+gem 'bootsnap', '>= 1.4.2', require: false
```
Comment (zspencer) [inline convene-web/Gemfile:67]: Apologies for the noise here, I wanted to group the gems by their purpose. I probably should document that in this file.

File: convene-web/app/controllers/features_controller.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+  end
+
+  helper_method def feature
+    @feature ||= Feature.find_by!(id: params[:id].to_sym)
+  end
+
+  helper_method def features
+    @features ||= Feature.all
+  end
```
Comment (zspencer) [inline convene-web/app/controllers/features_controller.rb:16]: I tend to prefer to use [decent_exposure](https://github.com/hashrocket/decent_exposure) for exposing models to the view; as it helps keep controllers light; but I am content to wait a bit.

**Comment by zspencer** [comment]
I realized after closing my laptop that this is probably a *Guide* and not a *Feature* and I think I want to rename before merge, so I am gonna convert this to a draft for now.

File: convene-web/app/views/features/index.html.erb (hunk 2 above)
Comment (anaulin) [reply convene-web/app/views/features/index.html.erb:5]: Never seen that gem before, but it seems like it does what we need and is not overly complicated, so it seems reasonable to use.

**Comment by anaulin** [comment]
Yeah, looking at https://convene-pr-176.herokuapp.com/features, agreed that it should probably say Guide and not Feature. Also it feels like we have some styling to do (though perhaps in a different PR, leave this one focused on the content).

**Review by user512 — APPROVED** [review]

**Pull request merged**